### DISCUSSION TITLE
shortVersionString ignored for current version

### DISCUSSION THREAD

**Discussion by dsmurfin** [sparkle-project/Sparkle#2323] [Q&A] [answered]
In our app we have the following:

`CFBundleVersion` single digit incrementing build - 1, 2, 3 etc.
`CFBundleShortVersionString` used facing version number e.g. `1.3.1`

In the app feed we have:

`sparkle:version` - matches CFBundleVersion
`sparkle:shortVersionString` - marketing version e.g. `1.3.2 Beta 2`

When Sparkle presents a dialog, we are not always getting the `shortVersionString`. For example (when running beta 1):

`XXX 1.3.2 Beta 2 is now available you - you have 1.3.2.` - why is shortVersionString not being used for current?

The same applies to `you're upto date XXX 1.3.2 is currently the newest version available` - why does this not say `XXX 1.3.2 Beta 2`

**Comment by zorgiepoo** [comment, marked as answer]
`CFBundleShortVersionString` from the app bundle is used when displaying the version of the app the user is currently using. It is usually assumed the `CFBundleShortVersionString` and `sparkle:shortVersionString` are used the same way.

Relatedly, I just merged some changes that allows better customization of the version strings #2321

**Comment by dsmurfin** [comment]
Thanks, really appreciate the info. I'm taking a look at your changes, and I think that might help.

**Comment by dsmurfin** [comment]
@zorgiepoo I am calling this from Swift, and the new functions do not seem to be being called (it is always the deprecated one). They are also not offered by auto-complete.

`formatUpdateDisplayVersion(from update: SUAppcastItem, bundleDisplayVersion: AutoreleasingUnsafeMutablePointer<NSString?>?, bundleVersion: String) -> String?`

`formatBundleDisplayVersion(_ bundleDisplayVersion: String, withBundleVersion bundleVersion: String, matchingUpdate: SUAppcastItem?) -> String`

**Reply by zorgiepoo** [reply]
I haven't made a release or published a beta for 2.4 yet. You are probably using an older version of Sparkle if you're not trying out a nightly build or building Sparkle yourself on top of tree.

**Reply by dsmurfin** [reply]
> I haven't made a release or published a beta for 2.4 yet. You are probably using an older version of Sparkle if you're not trying out a nightly build or building Sparkle yourself on top of tree.

Oh, I am using SPM from the PR commit you made into 2.x branch. The new API is clearly visible in the packages, but not being picked up by code completion.

**Reply by zorgiepoo** [reply]
That probably won’t work. We ship binary packages distributions for sparkle via SPM, not build from source ones. So we need to create a tag. Probably will this weekend for a beta of 2.4.

**Reply by dsmurfin** [reply]
Got it. I should have realised that, makes sense. The new API looks good thank you!

**Reply by zorgiepoo** [reply]
Just created a release for 2.4.0-beta.1 including these changes.

**Reply by dsmurfin** [reply]
Thank you. I'll give it a whirl.
